### PR TITLE
Created Blower Harness

### LINKED ISSUE #1289: Wire Harness: Blower
Create the blower harness, which essentially means putting the appropriate connectors onto the wires of the blower.

- Origin: PCB J54 Mega-Fit 2x1 and J39 Molex Micro-Fit 3.0 1x4
- Destination: Blower TKFAN BA7050H12B (other blowers may also be used in the future, but lets start with this one)
- Wire recommendation: use the wire already attached to the blower.
- Length: included length of blower wires.
- Pinout references: PCB schematic, blower datasheet (attached pdf to this ticket) [TENKAI TKFAN BA7050H12B.pdf](https://github.com/RespiraWorks/Ventilator/files/9137809/TENKAI.TKFAN.BA7050H12B.pdf)

A tutorial is available in our wiki here: https://github.com/RespiraWorks/Ventilator/wiki/Designing-Wire-Harnesses-with-WireViz

- [ ] Create a branch for the new harness
- [ ] Add components to the json library at: master/docs/purchasing/parts.json
- [ ] Create the harness source file using WireViz
- [ ] Run WireViz to check your work
- [ ] Create a documentation page for the harness in reStructured text
- [ ] Commit only the .yml and .rst files, and any images you references in the harness source file.
- [ ] Create a pull request for the harness branch

### PR DESCRIPTION
**Pull request by inceptionev**
# Description

- Created blower harness for TKFAN blower

Closes #1289 

## Self-review checklist:

- [x] Self-review: looked through the `Files changed` tab, browsed repository in branch
- [x] Documentation updated - reflects changes in code, electrical or mechanical design
- [x] Documentation and graphics follow the [documentation style guide](https://github.com/RespiraWorks/Ventilator/wiki/Documentation-style-guide)
- [x] New content is linked, easily discoverable, does not require too many clicks
- [x] Follows other relevant parts of [contributor wiki](https://github.com/RespiraWorks/Ventilator/wiki)
- [x] PR has a descriptive name
- [x] Tagged relevant reviewers

### Software only:

- [ ] Follows our [code style](https://github.com/RespiraWorks/Ventilator/wiki/Code-style)
- [ ] Commented code, particularly in hard-to-understand areas
- [ ] All tests pass - new and old, locally and on CI
- [ ] No new warnings or static check failures
- [ ] Tests that prove fix is effective or new feature works
- [ ] Manual tests are explained, with instructions for reproducing them

### CAD / mechanical only:

- [ ] Follows our [mechanical engineering standards](https://github.com/RespiraWorks/Ventilator/wiki/Mechanical-engineering-standards)
- [x] Everything is metric and ISO where possible
- [x] All features and constraints are named
- [x] Imported parts are well-named, with source URL and iProperties
- [ ] Screenshots and/or assembly photos updated

### CODE REVIEW

File: docs/purchasing/parts.json [hunk 1]
```diff
@@ -101,6 +101,57 @@
...
+    "mpn": "BA7050H12B",
+    "avg_price": 35.00,
+    "price_unit": "",
+    "sources": [
+      {
+        "supplier": "TKFAN",
+        "spn": "BA7050H12B",
+        "price": 35.00,
+        "price_units": 1,
+        "url": "http://tkfan.com/Hot_Sell/ba7050-22.html"
```
Comment (martukas) [inline docs/purchasing/parts.json:151]: Link here appears to be to the 24V version, which is not the one we are using?
Even if there is a page for the right version on the manufacturer's site, we should probably include a link to the one that can be bought on AliExpress, as that is practically where we got ours from, right?

File: docs/wiring/blower.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  :width: 800
+  :alt: Wiring diagram
+
+You will need all the tools listed.
+
+* Using items from above BOM as well as:
+   * 3mm heat shrink tubing [:ref:`ew-shrink3`]
+   * 5mm heat shrink tubing [:ref:`ew-shrink5`]
+   * Heat gun
+   * Molex Micro-Fit 3.0 Crimper
```
Comment (martukas) [inline docs/wiring/blower.rst:22]: Maybe we can have this tooling item in our database and link to it as well?

File: docs/wiring/blower.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   * Wire Strippers
+   * Soldering iron
+   * Solder
+* Instructions:
+   #. Cut 3 pieces of the 5mm heat shrink, 15mm long each.
+   #. Thread the 5mm heat shrink onto the the entire bundle of 4 wires on the blower.  Do not shrink the heat shrink yet.
+   #. Insert the 2x pre-crimped Mega-Fit wires into the Megafit 2x2 housing.
+   #. Do a gentle pull test to confirm proper seating of the contacts.
+   #. Cut the cable at a length of 50mm from the connector.
```
Comment (martukas) [inline docs/wiring/blower.rst:31]: Is it easier to insert in housing and then cut and solder, or solder first and insert in housing later?

File: docs/wiring/blower.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+   #. Strip 5mm of insulation from the end of the pre-crimped wires and from the ends of the red and black pigtails of the blower.
+   #. Cut 2 pieces of 3mm heat shrink, 15mm long.
+   #. Slide the 2 pieces of heat shrink onto the red and black wires of the blower.
```
Comment (martukas) [inline docs/wiring/blower.rst:34]: I would maybe phrase it like "insert each piece of shrink tubing onto each one of the red and black wires" or something like that? It seemed a bit ambiguous and one might insert both on the both wires together.

File: docs/wiring/blower.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+   #. Solder the red and black wires to the Mega-Fit connector cables according to the diagram (pin 1 on the Mega-Fit connector is indicated by a stripe on the connector body).
+   #. Slide the heat shrink over the soldered portion and use the heat gun to shrink them in place.
+   #. Strip the insulation from the blue pigtail of the blower, using the Micro-Fit crimp contact as a guide for the legth of insulation to strip.
+   #. Using the micro-fit crimper, crimp the contact onto the end of the blue wire.
+   #. Insert the contact and wire into the micro-fit connector housing according to the diagram.
+   #. Do a gentle pull test to confirm proper seating of the contacts.
+   #. Distribute the 5mm heat shrink tubing evenly along the harness and shrink them with the heat gun.
+
```
Comment (martukas) [inline docs/wiring/blower.rst:42]: I would say something about what to do with the yellow wire? I cut the exposed contact off and tucked it into one of the omnibus shrink tube sections to seal it away from accidentally contacting something.

**Review by martukas — CHANGES_REQUESTED** [review]
Fundamentally this is good. I have successfully assembled this and tested it with the bring-up `ino` code to confirm that the blower gets power and signal.

The only salient thing that I think we need to change before merging this is a link to the appropriate model of blower and where it can be bought.

The other points are minor improvements to the instructions.

As a knock-on of this review, a few issues have been identified with reproducibility of the bring-up testing, as outlined in #1321

File: docs/wiring/blower.rst (hunk 2 above)
Comment (inceptionev) [reply docs/wiring/blower.rst]: Sure, I've added it to the database.  I also created a new category in the parts schema to differentiate tools (which should be bought once) from parts (which are bought per assembly).

File: docs/wiring/blower.rst (hunk 3 above)
Comment (inceptionev) [reply docs/wiring/blower.rst:31]: I think the wires are long enough that either way is fine.  Did you have trouble with this?

File: docs/wiring/blower.rst (hunk 4 above)
Comment (inceptionev) [reply docs/wiring/blower.rst:34]: done.

File: docs/wiring/blower.rst (hunk 5 above)
Comment (inceptionev) [reply docs/wiring/blower.rst:42]: done.

**Review by martukas — APPROVED** [review]
LGTM

**Pull request merged**